larpandoracontent/LArObjects/LArDiscreteProbabilityVector.cc

```diff
@@ -0,0 +1,237 @@
+/**
+ *  @file   larpandoracontent/LArObjects/LArDiscreteProbabilityVector.cc
+ *
+ *  @brief  Implementation of the lar discrete probability vector class
+ *
+ *  $Log: $
+ */
+
+#include "larpandoracontent/LArObjects/LArDiscreteProbabilityVector.h"
+
+#include <algorithm>
+#include <cmath>
+#include <numeric>
+
+namespace lar_content
+{
+
+template <typename TX, typename TY>
+DiscreteProbabilityVector::DiscreteProbabilityVector(const InputData<TX, TY> &inputData, const TX xUpperBound, 
+    const bool useWidths) :
+    m_xUpperBound(static_cast<float>(xUpperBound)),
+    m_useWidths(useWidths),
+    m_discreteProbabilityData(this->InitialiseDiscreteProbabilityData(inputData))
+{
+    this->VerifyCompleteData();
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+DiscreteProbabilityVector::DiscreteProbabilityVector(const DiscreteProbabilityVector &discreteProbabilityVector,
+    std::mt19937 &randomNumberGenerator) :
+    m_xUpperBound(discreteProbabilityVector.m_xUpperBound),
+    m_useWidths(discreteProbabilityVector.m_useWidths),
+    m_discreteProbabilityData(this->RandomiseDiscreteProbabilityData(discreteProbabilityVector, randomNumberGenerator))
+{
+    this->VerifyCompleteData();
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+DiscreteProbabilityVector::DiscreteProbabilityVector(const DiscreteProbabilityVector &discreteProbabilityVector,
+    const ResamplingPoints &resamplingPoints) :
```

Comment:
Extra 4 spaces indent for constructor arguments.

Reply:
Got it in a previous comment